### PR TITLE
Prometheus: Fix annotation query request for browser access mode

### PR DESCRIPTION
**Pull request by itsmylife**
**What is this feature?**
Only for `v9.1.x` we are checking access more before making an annotation query. After `v9.1.x` we removed the direct access from prometheus datasource

**Which issue(s) does this PR fix?**:
https://github.com/grafana/grafana/issues/55705

Fixes https://github.com/grafana/grafana/issues/55705

**Special notes for your reviewer**:
I had to use `any` for `filter` pipe. I couldn't provide a happy type for it. I could use some help over there.

### CODE REVIEW

File: public/app/plugins/datasource/prometheus/datasource.tsx [hunk 1]
```diff
@@ -726,6 +727,34 @@ export class PrometheusDatasource
...
+      const queryModel = {
+        expr,
+        interval: step,
+        refId: 'X',
+        requestId: `prom-query-${annotation.name}`,
+      };
+      const query = this.createQuery(queryModel, queryOptions, start, end);
+      return await lastValueFrom(
+        this.performTimeSeriesQuery(query, query.start, query.end).pipe(
+          filter((response: any) => isFetchSuccessResponse(response)),
```
Comment (itsmylife) [inline public/app/plugins/datasource/prometheus/datasource.tsx:744]: 😢 Help requested for the type

Comment (bohandley) [reply public/app/plugins/datasource/prometheus/datasource.tsx:744]: This one is challenging! I was unable to figure it out.

Comment (ivanahuckova) [reply public/app/plugins/datasource/prometheus/datasource.tsx:744]:
```suggestion
          filter(isFetchSuccessResponse),
```

I think this should work. I've found this good blog post on it https://medium.com/ngconf/filtering-types-with-correct-type-inference-in-rxjs-f4edf064880d

Comment (itsmylife) [reply public/app/plugins/datasource/prometheus/datasource.tsx:744]: @ivanahuckova that is worked. But I tried that while I was working on this and it failed. I must have missed something. Thanks!

**Review by obetomuniz — APPROVED** [review]
LGTM

**Comment by obetomuniz** [comment]
Hey @itsmylife, could you check which label will be defined to address `changelog`? If we remain with `add to changelog`, I believe you need to add additional details in the PR description.

**Pull request merged**